### PR TITLE
Adding unit tests for crd validation

### PR DESCRIPTION
**Pull request by sbezverk**
This PR adds unit tests for CRD validation. Since the actual validation is done by the server, it must run against a live k8s cluster as a result `--kube-config` parameter must be passed to test:
Example: `go test -v --kube-config=/Users/sbezverk/.kube/config`

These tests are targeted for new CI environment, where a live k8s cluster gets build.
 
Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

### CODE REVIEW

**Comment by sbezverk** [comment]
```
SBEZVERK-M-L0L4:crd sbezverk$ go test -v --kube-config=/Users/sbezverk/.kube/config
=== RUN   TestCRDValidation
--- PASS: TestCRDValidation (0.09s)
PASS
ok      github.com/ligato/networkservicemesh/plugins/crd        0.123s
```

**Comment by sbezverk** [comment]
@fkautz @mestery  PTAL

File: plugins/crd/plugin_impl_crd.go [hunk 1]
```diff
@@ -126,7 +125,7 @@ func networkServiceValidation() *apiextv1beta1.CustomResourceValidation {
 							Type:        "string",
 							MaxLength:   &maxLength,
 							Description: "NetworkService Name",
-							Pattern:     `^[a-zA-Z0-9]+\-[a-zA-Z0-9]*$`,
+							Pattern:     `^[a-zA-Z0-9]+[\-a-zA-Z0-9]*$`,
```
Comment (mestery) [inline plugins/crd/plugin_impl_crd.go]: Is this change related, or did the unit test catch this? Same for lines 156, 182 and 196.

File: plugins/crd/plugin_impl_crd.go [hunk 2]
```diff
@@ -227,16 +226,7 @@ func createCRD(plugin *Plugin, FullName, Group, Version, Plural, Name string) er
 			Validation: validation,
 		},
 	}
-
 	_, cserr := plugin.apiclientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
-	if cserr != nil && apierrors.IsAlreadyExists(cserr) {
```
Comment (mestery) [inline plugins/crd/plugin_impl_crd.go]: Why remove these printouts?

File: plugins/crd/plugin_impl_crd_test.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	"flag"
+	"reflect"
+	"testing"
+
+	"github.com/ligato/networkservicemesh/netmesh/model/netmesh"
+	"github.com/ligato/networkservicemesh/pkg/apis/networkservicemesh.io/v1"
+	networkservicemesh "github.com/ligato/networkservicemesh/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	apiextcs "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (mestery) [inline plugins/crd/plugin_impl_crd_test.go]: The pattern for importing meta/v1 has been to simply name the import "meta".

File: plugins/crd/plugin_impl_crd_test.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		t.Skip("This test requires a valid kubeconfig file, skipping...")
+	}
+	k8sClient, apiextClient, crdClient, err := k8sClient(kubeconfig)
+	if err != nil {
+		t.Errorf("Fail to get k8s client with error: %+v", err)
+	}
+	if err := setupEnv(k8sClient, apiextClient); err != nil {
+		t.Errorf("Fail to setup test environment with error: %+v", err)
+	}
+	defer func() {
```
Comment (mestery) [inline plugins/crd/plugin_impl_crd_test.go]: Love the usage of defer() here, it's one of the things I like about go. :)

File: plugins/crd/plugin_impl_crd.go (hunk 1 above)
Comment (sbezverk) [reply plugins/crd/plugin_impl_crd.go]: Running new unit test I found a bug in original regex expression, so it is a new corrected pattern.

File: plugins/crd/plugin_impl_crd.go (hunk 2 above)
Comment (sbezverk) [reply plugins/crd/plugin_impl_crd.go]: crdCreate function should not print anything but just return the result, by doing fancy printing using `plugin` logger method, breaks unit testing as it is really hard to initialize logger.

File: plugins/crd/plugin_impl_crd_test.go (hunk 3 above)
Comment (sbezverk) [reply plugins/crd/plugin_impl_crd_test.go]: will change

File: plugins/crd/plugin_impl_crd_test.go (hunk 4 above)
Comment (sbezverk) [reply plugins/crd/plugin_impl_crd_test.go]: Yeah, I really like it too!!

File: scripts/integration-tests [hunk 5]
```diff
@@ -45,6 +45,7 @@ kubectl get nodes
 kubectl version
 kubectl api-versions
 kubectl label --overwrite --all=true nodes app=networkservice-node
+
```
Comment (mestery) [inline scripts/integration-tests]: Remove this blank line.

File: scripts/integration-tests [hunk 6]
```diff
@@ -64,6 +65,10 @@ kubectl create -f conf/sample/networkservice-channel.yaml
 kubectl create -f conf/sample/networkservice-endpoint.yaml
 kubectl create -f conf/sample/networkservice.yaml
 kubectl get pod,crd,NetworkService,NetworkServiceEndpoint,NetworkServiceChannel --all-namespaces
+
+# Need to get kubeconfig full path
+k8sconfig=$HOME'/.kube/config'
```
Comment (mestery) [inline scripts/integration-tests]: Lets keep variables capitalized here, and also correct the quotes. Change this to:

`K8SCONFIG=$HOME/.kube/config`

File: scripts/integration-tests [hunk 7]
```diff
@@ -64,6 +65,10 @@ kubectl create -f conf/sample/networkservice-channel.yaml
...
+go test ./plugins/crd/... -v --kube-config=$k8sconfig
```
Comment (mestery) [inline scripts/integration-tests]: And then subsequently update this to:

`go test ./plugins/crd/... -v --kube-config=$K8SCONFIG`

**Comment by sbezverk** [comment]
@mestery Addressed requested changes...
